From f04c1e4b98451adf2533545c929679862b8240fb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Jirman?= <megi@xff.cz>
Date: Wed, 5 Aug 2020 11:19:01 +0200
Subject: [PATCH 093/391] arm64: dts: sun50i-a64-pinephone: Fix BH modem
 manager behavior

PP 1.1 has STATUS line multiplexed with POWER KEY, but reading it doesn't
work reliably. Just treat 1.1 as if it was 1.0, and ignore the STATUS.
---
 arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.1.dts | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.1.dts b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.1.dts
index 93dc8512f..d67f31ef5 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.1.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.1.dts
@@ -123,8 +123,8 @@ modem {
 
 		enable-gpios = <&pio 7 8 GPIO_ACTIVE_LOW>; /* PH8 */
 		reset-gpios = <&pio 2 4 GPIO_ACTIVE_HIGH>; /* PC4 */
-		status-gpios = <&pio 1 3 GPIO_ACTIVE_HIGH>; /* PB3 */
-		status-pwrkey-multiplexed; /* status acts as pwrkey */
+		pwrkey-gpios = <&pio 1 3 GPIO_ACTIVE_HIGH>; /* PB3 */
+		//status-pwrkey-multiplexed; /* status acts as pwrkey */
 
 		sleep-gpios = <&pio 7 7 GPIO_ACTIVE_HIGH>; /* PH7 */
 		wakeup-gpios = <&pio 1 2 GPIO_ACTIVE_HIGH>; /* PB2-RI */
-- 
2.35.3

